### PR TITLE
fix: remove RapidAPI plugin src code from API server

### LINKED ISSUE #16440: [Task]: Remove RapidAPI plugin code from the API server
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

- At the moment we do not have plans to support RapidAPI integration and hence it is necessary to remove the related plugin code from the API server so that it doesn't lead to tech debt or force developers to consider its presence.

### PR DESCRIPTION
**Pull request by sumitsum**
## Description
- Remove RapidAPI plugin src code from the API server since we are not supporting the plugin at the moment. 
- This PR does the following: 
  - mark the plugin as `deleted` in the `plugin` collection.
  - mark the plugin as `deleted` in the `workspace` collection.
  - mark the datasources related to the plugin as `deleted`.
  - mark the related actions as `deleted`
  - remove the plugin from the `pom.xml` file of `appsmith-plugins` module
  - when reading the list of installed plugins in a workspace only select those plugins that are not marked as `deleted`
  - remove code that adds the plugin to the `plugin` collection
- Please note that everywhere only `soft` delete happens during the data migration phase. This is done for two reasons: 
  - it makes it possible to recover older state since data would still exist.
  - no change in indexing in case any index has been built upon any collection.

Fixes #16440 

## Type of change
- Tech debt

## How Has This Been Tested?
- Manual
- Automated test does not seem feasible for this change.

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog.java [hunk 1]
```diff
@@ -288,18 +288,6 @@ public void initialPlugins(MongockTemplate mongoTemplate) {
         } catch (DuplicateKeyException e) {
             log.warn("mongo-plugin already present in database.");
         }
-
```
Comment (mohanarpit) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog.java:291]: As a principle, we shouldn't modify older database changelogs unless necessary. Is removing this code required?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog2.java [hunk 2]
```diff
@@ -2435,4 +2436,104 @@ public void updateBadThemeState(MongockTemplate mongockTemplate, @NonLockGuarded
...
+    @ChangeSet(order = "035", id = "delete-rapid-api-plugin-related-items", author = "")
+    public void deleteRapidApiPluginRelatedItems(MongockTemplate mongockTemplate) {
+        Plugin rapidApiPlugin = mongockTemplate.findOne(query(where("packageName").is("rapidapi-plugin")),
+                Plugin.class);
+
+        if (rapidApiPlugin == null) {
+            return;
+        }
+
+        softDeleteAllRapidApiActions(rapidApiPlugin, mongockTemplate);
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog2.java:2448]: Nit: Can we make this into a generic method so migrations for deleting plugins are easier in the future?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog2.java [hunk 3]
```diff
@@ -2435,4 +2436,104 @@ public void updateBadThemeState(MongockTemplate mongockTemplate, @NonLockGuarded
...
+        softDeleteAllRapidApiDatasources(rapidApiPlugin, mongockTemplate);
+        softDeleteRapidApiPluginFromAllWorkspaces(rapidApiPlugin, mongockTemplate);
+        softDeletePlugin(rapidApiPlugin, mongockTemplate);
+    }
+
+    private void softDeleteRapidApiPluginFromAllWorkspaces(Plugin rapidApiPlugin, MongockTemplate mongockTemplate) {
+        Query queryToGetNonDeletedWorkspaces = new Query(where("deleted").ne(true));
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog2.java:2455]: If we are doing soft deletes for everything, would it makes sense to perform these migrations for soft deleted workspaces as well?

**Review by nidhi-nair — COMMENTED** [review]
Minor comments

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog.java (hunk 1 above)
Comment (sumitsum) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog.java]: @mohanarpit agreed, I have removed this change. Initially I thought that maybe we should stop the installation of the plugin itself. However, removing the code from database changelog does seem like an anti-pattern.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog2.java (hunk 3 above)
Comment (sumitsum) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog2.java]: Done.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog2.java (hunk 2 above)
Comment (sumitsum) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog2.java]: Done.

**Review by nidhi-nair — DISMISSED** [review]

**Comment by sumitsum** [comment]
> @sumitsum just noticed that there are a couple of RapidAPI related properties in the `application.properties` file. Could you please check if we need those?

@nidhi-nair thanks for pointing this out. I have updated the PR to remove the related properties.

**Review by nidhi-nair — APPROVED** [review]

**Pull request merged**